web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -20,13 +20,13 @@
         <div class="header branch-tag-choice">
           <div class="ui grid">
             <div class="two column row">
-              <a class="reference column" href="#" @click="createTag = false; mode = 'branches'; focusSearchField()">
+              <a class="reference column" href="#" @click="createTag = false; mode = 'branches'; focusSearchField(); fetchBranches()">
                 <span class="text" :class="{black: mode === 'branches'}">
                   <svg-icon name="octicon-git-branch" :size="16" class-name="gt-mr-2"/>{{ textBranches }}
                 </span>
               </a>
               <template v-if="!noTag">
-                <a class="reference column" href="#" @click="createTag = true; mode = 'tags'; focusSearchField()">
+                <a class="reference column" href="#" @click="createTag = true; mode = 'tags'; focusSearchField(); fetchBranches()">
```

Comment:
These two `reference column` elements need their own function now.

Reply:
Same as above. Now the goal of this PR is to send a request to get all branches using restful api and use loading icon while waiting for the response (like github).

Reply:
But still need to test the performance change first.

Reply:
added the handler